internalshared/configutil/config.go

```diff
@@ -0,0 +1,211 @@
+package configutil
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+	"github.com/hashicorp/vault/sdk/helper/parseutil"
+)
+
+// SharedConfig contains some shared values
+type SharedConfig struct {
+	EntSharedConfig
+
+	Listeners []*Listener `hcl:"-"`
+
+	Seals   []*KMS   `hcl:"-"`
+	Entropy *Entropy `hcl:"-"`
+
+	DisableMlock    bool        `hcl:"-"`
+	DisableMlockRaw interface{} `hcl:"disable_mlock"`
+
+	Telemetry *Telemetry `hcl:"telemetry"`
+
+	DefaultMaxRequestDuration    time.Duration `hcl:"-"`
+	DefaultMaxRequestDurationRaw interface{}   `hcl:"default_max_request_duration"`
+
+	// LogFormat specifies the log format. Valid values are "standard" and
+	// "json". The values are case-insenstive. If no log format is specified,
+	// then standard format will be used.
+	LogFormat string `hcl:"log_format"`
+	LogLevel  string `hcl:"log_level"`
+
+	PidFile string `hcl:"pid_file"`
+
+	ClusterName string `hcl:"cluster_name"`
+}
+
+// LoadConfigFile loads the configuration from the given file.
+func LoadConfigFile(path string) (*SharedConfig, error) {
+	// Read the file
+	d, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return ParseConfig(string(d))
+}
+
+func ParseConfig(d string) (*SharedConfig, error) {
+	// Parse!
+	obj, err := hcl.Parse(d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Start building the result
+	var result SharedConfig
+	if err := hcl.DecodeObject(&result, obj); err != nil {
+		return nil, err
+	}
+
+	if result.DefaultMaxRequestDurationRaw != nil {
+		if result.DefaultMaxRequestDuration, err = parseutil.ParseDurationSecond(result.DefaultMaxRequestDurationRaw); err != nil {
+			return nil, err
+		}
+		result.DefaultMaxRequestDurationRaw = nil
+	}
+
+	if result.DisableMlockRaw != nil {
+		if result.DisableMlock, err = parseutil.ParseBool(result.DisableMlockRaw); err != nil {
+			return nil, err
+		}
+		result.DisableMlockRaw = nil
+	}
+
+	list, ok := obj.Node.(*ast.ObjectList)
+	if !ok {
+		return nil, fmt.Errorf("error parsing: file doesn't contain a root object")
+	}
+
+	if o := list.Filter("hsm"); len(o.Items) > 0 {
+		if err := parseKMS(&result, o, "hsm", 1); err != nil {
+			return nil, errwrap.Wrapf("error parsing 'hsm': {{err}}", err)
+		}
+	}
+
+	if o := list.Filter("seal"); len(o.Items) > 0 {
+		if err := parseKMS(&result, o, "seal", 2); err != nil {
+			return nil, errwrap.Wrapf("error parsing 'seal': {{err}}", err)
+		}
+	}
+
+	if o := list.Filter("kms"); len(o.Items) > 0 {
```

Comment:
I noticed that we're introducing this new stanza, `kms`. Are there plans to deprecate `seal`?

Reply:
Ideally, but for now seal still works. Really it's that seal is both a Vault-specific name and isn't actually descriptive of what it does, so for other users of this code I want to allow `kms` from the get-go.